pages/index.js

```diff
@@ -1,68 +1,67 @@
-import envFlag from 'node-env-flag';
 import React from 'react';
-import Meta from '../frontend/components/meta';
-import Header from '../frontend/components/header';
-import SuggestionAndSearch from '../frontend/components/suggestion-and-search';
+import { HashRouter, StaticRouter, Route, Link } from "react-router-dom";
 import { BadgeExamples } from '../frontend/components/badge-examples';
-import MarkupModal from '../frontend/components/markup-modal';
-import Usage from '../frontend/components/usage';
-import Footer from '../frontend/components/footer';
-import badgeExampleData from '../badge-examples.json';
-import { prepareExamples, predicateFromQuery } from '../frontend/lib/prepare-examples';
+import ExamplesPage from '../frontend/components/examples-page';
+import { baseUri, longCache } from '../frontend/constants';
 
-const baseUri = process.env.BASE_URL;
-const longCache = envFlag(process.env.LONG_CACHE, false);
-
-export default class IndexPage extends React.Component {
-  state = {
-    query: null,
-    example: null
-  };
+class IndexPage extends ExamplesPage {
 
   constructor(props) {
     super(props);
-    this.preparedExamples = prepareExamples(
-      badgeExampleData,
-      () => predicateFromQuery(this.state.query));
+    this.state.searchReady = false;
+    this.searchTimeout = 0;
   }
 
-  render() {
-    return (
-      <div>
-        <Meta />
-        <Header />
-        <MarkupModal
-          example={this.state.example}
-          onRequestClose={() => { this.setState({ example: null }); }}
-          baseUri={baseUri} />
-        <section>
-          <SuggestionAndSearch
-            queryChanged={query => { this.setState({ query }); }}
-            onBadgeClick={example => { this.setState({ example }); }}
-            baseUri={baseUri}
-            longCache={longCache} />
-          <a
-            className="donate"
-            href="https://opencollective.com/shields">
-            donate
-          </a>
-        </section>
-        <BadgeExamples
+  getBody() {
+    if ((this.state.query == null) || (this.state.query.length === 0)) {
+      return this.preparedExamples.map(function(category, i) {
+        return (
+          <Link to={'/examples/' + category.category.id} key={i}>
+            <h3 id={category.category.id}>{ category.category.name }</h3>
+          </Link>
+        )
+      });
+    } else if (this.state.query.length === 1) {
+      return (
+        <div>Search term must have 2 or more characters</div>
```

Comment:
Because this PR removes rows from the DOM rather than hiding them – by design of course! – it seems like we could be a little more conservative about when we initiate a search. If users want to browse, they can click the sections instead of searching. It seems like 3 would be enough… git, npm, etc. What do you think?

Reply:
I don't think having the minimum set to 2 should be too much of a problem,
The timeout between characters is set to 500ms and the average person types at ~300ms per character.
>[The average person types between 38 and 40 words per minute -- between 190 and 200 characters per minute.](https://www.google.co.nz/search?q=average+cpm+typing+speed)

So most of the time, there should be no difference between having the minimum as 2/3 if the user does type `>=3` characters.